test/runtime/ct.go

```diff
@@ -619,15 +619,20 @@ var _ = Describe("RuntimeValidatedConntrackTable", func() {
 		res := vm.SetPolicyEnforcement(helpers.PolicyEnforcementDefault)
 		res.ExpectSuccess("Setting policy enforcement as default")
 
+		areEndpointsReady := vm.WaitEndpointsReady()
+		Expect(areEndpointsReady).Should(BeTrue(), "Endpoints are not ready after timeout")
+
 		// #FIXME remove these 6 lines once GH-2496 is fixed
 		epIDs, err := vm.GetEndpointsIds()
 		Expect(err).To(BeNil(), "Getting endpoints identity IDs")
 		for _, v := range epIDs {
-			vm.ExecCilium(fmt.Sprintf("endpoint config %s IngressPolicy=false", v)).ExpectSuccess("Setting %s endpoint's IngressPolicy as false", v)
-			vm.ExecCilium(fmt.Sprintf("endpoint config %s EgressPolicy=false", v)).ExpectSuccess("Setting %s endpoint's EgressPolicy as false", v)
+			vm.ExecCilium(fmt.Sprintf("endpoint config %s IngressPolicy=false", v)).ExpectSuccess(
+				"Setting %s endpoint's IngressPolicy as false", v)
```

Comment:
https://github.com/cilium/cilium/pull/3054 was merged to unblock the CI. What I think should be done to fix this issue once the correct functionality that was actually reverted in https://github.com/cilium/cilium/pull/3054 is restored, is after each configuration update for an endpoint, wait until the endpoint is regenerated. The only wrinkle there though, is sometimes the endpoint does not actually have to be regenerated :) I think we can get around this by checking the output of `cilium endpoint config` for each endpoint and then if the option does need to be changed, change it and then wait for regeneration.

Reply:
After each configuration update, add: 
```
areEndpointsReady = vm.WaitEndpointsReady()
Expect(areEndpointsReady).Should(BeTrue(), "Endpoints not ready after timeout")
```

I think we should eventually factor config updates for endpoints / cilium into helper functions which wait for endpoints to be in ready state after applying such updates.